### PR TITLE
Fix bugs introduced with post-v.3.3 updates to LXC codebase

### PR DESCRIPTION
**Pull request by emaballarin**
Hello! After trying - and failing - to install the package, post-update to `pkgrel=2` (which isn't the root cause, it just happened to occur at the same time as some changes in the upstream lxc/lxd repository), I noticed that at least two other people are affected by the same problem(s) as mine, as per the AUR comment section.

I tried to fix the issue(s) in the most concise way possible, by adding some lines to the `build` section of the `PKGBUILD`. I have marked them with comments for repid erasure should the "bugs" (at least one of them) be fixed upstream. However, they should cause no harm even in that case.

I hope they could be of some usefulness.

Thank you in advance.

---

In detail:
- The first fix is just the implementation of the batch of suggestions given by the build script, at the series of errors ensuing in trying to fix them sequentially;
- The second one tries to overcome the known issue with the `xattrs.h` header which was usually located at `<system include dir>/attrs/` but, after a breaking version upgrade, was moved to `<system include dir>/sys/`. An update to the lxc codebase a couple of days ago introduced the library as a dependency, thus causing the error.

### CODE REVIEW

**Comment by lenucksi** [comment]
I was also able to reproduce the issue with libsqlite3 on upgrade to PKGREL 2. Thanks for making a PR for this!

**Comment by monstermunchkin** [comment]
The issue with `xattr.h` has been fixed upstream.

**Comment by emaballarin** [comment]
@monstermunchkin Thanks for the fix (and for the whole LXC infrastructure, btw)!

File: PKGBUILD [hunk 1]
```diff
@@ -9,8 +9,8 @@ arch=('x86_64')
 url="https://github.com/lxc/lxd"
 license=('APACHE')
 conflicts=('lxd-lts')
-depends=('lxc' 'squashfs-tools' 'dnsmasq' 'sqlite')
-makedepends=('go-pie' 'git')
+depends=('lxc' 'squashfs-tools' 'dnsmasq' 'sqlite' 'attr')
+makedepends=('go-pie' 'git' 'sed')
```
Comment (monstermunchkin) [inline PKGBUILD]: Just a few notes:
* `sed` is part of the group `base-devel` and therefore [should not be added to the array](https://wiki.archlinux.org/index.php/PKGBUILD#makedepends)
* `attr` isn't a dependency of LXD
* you should remove the `sqlite` dependency, and therefore add `dqlite`

File: PKGBUILD [hunk 2]
```diff
@@ -42,7 +42,15 @@ build() {
   go_base=github.com/lxc/lxd
   mkdir -p "${GOPATH}"
   GOPATH="${GOPATH}" go get "${go_base}" || echo "(ignoring go error)"
+  # Fix for error "Missing custom libsqlite3[...]" (pt.1/2)
+  export CGO_CFLAGS="-I${srcdir}/go/deps/sqlite/ -I${srcdir}/go/deps/dqlite/include/"
+  export CGO_LDFLAGS="-L${srcdir}/go/deps/sqlite/.libs/ -L${srcdir}/go/deps/dqlite/.libs/"
+  export LD_LIBRARY_PATH="${srcdir}/go/deps/sqlite/.libs/:${srcdir}/go/deps/dqlite/.libs/"
```
Comment (monstermunchkin) [inline PKGBUILD]: These can be removed when you add `dqlite` as a dependency.

File: PKGBUILD [hunk 3]
```diff
@@ -42,7 +42,15 @@ build() {
...
+  # Normal execution
   cd "${GOPATH}/src/${go_base}"
+  # Fix for error "Missing custom libsqlite3[...]" (pt.2/2)
+  make deps
```
Comment (monstermunchkin) [inline PKGBUILD]: Same as above.

File: PKGBUILD (hunk 1 above)
Comment (sjemens) [reply PKGBUILD]: The current release of dqlite is v0.1.0 and is missing a few things that are required from lxd, but adding the dqlite-git as dependency resolves those issues.

**Comment by emaballarin** [comment]
Thank you all for the useful suggestions! Now I have updated the PR accordingly (and I can confirm that I am able to build correctly with dqlite-git).

**Comment by maikelwever** [comment]
So this PR is now ready to merge?

**Pull request merged**